.github/workflows/ci.yml

```diff
@@ -30,7 +30,7 @@ jobs:
           cache: pnpm
       - run: pnpm install --frozen-lockfile
       - run: pnpm run lint
-      - run: cd packages/kit && pnpm prepublishOnly
+      - run: cd packages/kit && pnpm prepublishOnly && [ "`git status --porcelain=v1`" == "" ] # ensures that commited types are up-to-date
```

Comment:
If this fails, it might be hard for users to understand the failure. Would it work to put this as a `script` inside `package.json` like `"checkIfTypesUpdated": "[ \"`git status --porcelain=v1`\" == '' ]"`? Then they'd see that `pnpm checkIfTypesUpdated` failed, which might be a bit more informative

Reply:
would be nice to keep it as a one-liner. what about this?

```suggestion
      - run: cd packages/kit && pnpm prepublishOnly && { [ "`git status --porcelain=v1`" == "" ] || echo "Generated types have changed — please run prepublishOnly locally and commit the changes after you have reviewed them"; }
```